### PR TITLE
add token vault feature

### PR DESCRIPTION
**Pull request by psarna**
Adds token vault capabilities which translate auth tokens
on the fly, so that the proxees do not actually ever see
the real tokens.

Flow:
```sh
$ ./sandworm token add bearer --host api.github.com --secret "$GITHUB_API_TOKEN"
created bearer mapping "api.github.com" for api.github.com * using proxy token "default"
$ ./sandworm token list
default
  - bearer api.github.com *
```

```sh
$ ./sandworm --domains api.github.com --translate-tokens
time=2026-04-27T09:45:37.688+02:00 level=INFO msg="Starting proxy" addr=0.0.0.0:2137 mode=allow allowed_domains=[api.github.com] allowed_cidrs=[] blocked_domains=[] blocked_cidrs=[]
time=2026-04-27T09:45:37.688+02:00 level=INFO msg="Starting admin panel" addr=127.0.0.1:2138
time=2026-04-27T09:45:37.688+02:00 level=INFO msg="Proxy started successfully" port=2137 admin_bind=127.0.0.1 mode=allow domains=[api.github.com] cidrs=[] blocked_domains=[] blocked_cidrs=[] translate_tokens=true translate_tokens_db=./sandworm.db

time=2026-04-27T09:52:49.262+02:00 level=INFO msg="Proxy request" method=CONNECT host=api.github.com:443 url=//api.github.com:443 remote=127.0.0.1:58068
time=2026-04-27T09:52:59.189+02:00 level=INFO msg="Proxy request" method=CONNECT host=api.github.com:443 url=//api.github.com:443 remote=127.0.0.1:43258
time=2026-04-27T09:53:02.509+02:00 level=INFO msg="Proxy request" method=CONNECT host=api.github.com:443 url=//api.github.com:443 remote=127.0.0.1:54504
```

```sh
$ . <(./sandworm token config)
$ echo $HTTPS_PROXY
http://x:sand_-khhQ0jx0wOPxaoyGajlCdchfBWB-ZXu@127.0.0.1:2137

$ curl -sSfL https://api.github.com/user | jq -r '.name,.company'
Piotr Sarna
poolside
```

Revoking path:
```sh
$ ./sandworm token revoke
revoked proxy token "default"
$ ./sandworm token list
default
  - no mappings

$ curl -sSfL https://api.github.com/user | jq -r '.name,.company'
curl: (22) The requested URL returned error: 401

```

### CODE REVIEW

**Comment by yangchi** [comment]
O_O this is beyond my know how . I will take a look but it will be slow

(Update: looked. Still don't think I'm the right person to look. I have nothing useful to offer)

File: pkg/proxy/proxy.go [hunk 1]
```diff
@@ -63,9 +63,13 @@ type Proxy struct {
...
 	dnsPort   int
 	dnsServer *DNSServer
+
+	translationStore *TranslationStore
+	translationCA    *certificateAuthority
+	httpClient       *http.Client
 }
 
-func NewProxy(port int, adminBind string, mode string, allowedDomains []string, allowedCIDRs []string, blockedDomains []string, blockedCIDRs []string, ipProxyRange string, ipPorts []int, dnsPort int) *Proxy {
+func NewProxy(port int, adminBind string, mode string, allowedDomains []string, allowedCIDRs []string, blockedDomains []string, blockedCIDRs []string, ipProxyRange string, ipPorts []int, dnsPort int, tokenDB string) (*Proxy, error) {
```
Comment (bronsted) [inline pkg/proxy/proxy.go]: The list of arguments is becoming a bit long. Consider using an argument struct or an option pattern

**Review by bronsted — APPROVED** [review]
This is very cool.

A few places, it might be a good idea to add a few more tests touching some of the edge cases.

Lets try it!
